pkg/collectors/devices/fetcher.go

```diff
@@ -51,9 +51,44 @@ func (inst *fetcher) AddCommand(cmdInst *clients.Cmd) {
 	inst.cmdGrp.AddCommand(cmdInst)
 }
 
+func setValue(field *reflect.StructField, fieldVal reflect.Value, valueName string, value any) error {
+	//nolint:exhaustive //we could extend this but its not needed yet
+	switch field.Type.Kind() {
+	case reflect.String:
+
+		stringRes, ok := value.(string)
+		if !ok {
+			return fmt.Errorf("failed to convert %v into string", value)
+		}
+		fieldVal.SetString(stringRes)
+	case reflect.Slice:
+		resType := reflect.TypeOf(value)
+		if field.Type != resType {
+			return fmt.Errorf(
+				"type of %s (%v) does not match field type %v",
+				valueName,
+				resType,
+				field.Type,
+			)
+		}
+		if field.Type.Elem() != resType.Elem() {
+			return fmt.Errorf(
+				"InnerType of %s (%v) does not match field InnerType %v",
+				valueName,
+				resType.Elem(),
+				field.Type.Elem(),
+			)
+		}
+		fieldVal.Set(reflect.ValueOf(value))
+	default:
+		return fmt.Errorf("fetcher unmarshal not implemented for type: %s", field.Type.Name())
```

Comment:
Check which is the correct spelling of "unmarshal". This error message is different to the function name in a few lines.